### PR TITLE
Notification Entities + Swift Mk 1: Managed Object

### PR DESCRIPTION
**Pull request by jleandroperez**
#### Details:

Migrating **Notification** SPManagedObject. Reference #5613

Needs Review: @kwonye 
Getting there... clooooooose... clooooose!!

Thanks in advance!!!

#### Testing:
1. Validate that the Notifications List properly groups by date
2. Validate that the Notifications List Filters didn't break
3. Perform a smoke test over **Like** / **Follow** / **Comment** notifications, and check that nothing broke
4. Press over a Matcher notification, and validate that the native reader gets pushed
5. Run the Notifications Unit Tests, and verify everythign passes.

### CODE REVIEW

**Comment by kwonye** [comment]
@jleandroperez Travis is failing for a seemingly unrelated reason. Any ideas?

**Comment by kwonye** [comment]
Also, `[WPAvatarSourceTest testGravatarCaches]` is failing in this branch (it's fine in `develop`)

**Comment by jleandroperez** [comment]
@kwonye can't trigger the `[WPAvatarSourceTest testGravatarCaches]` failure locally, but since none of that code was updated / modified here, i dare to say it may be a weird timming glitch.

As per **Travis**, i usually just disregard, and verify locally. I (think) @nheagy has been working on fixing that lately? (Nate is that possible or i'm confused?).

I'll add another dummy commit to force Travis re-test. Thanks @kwonye !!

File: WordPress/Classes/Models/Notifications/Notification+CoreDataProperties.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import Foundation
+import CoreData
+
+
+// MARK: - Notification Core Data Properties
+//
+extension Notification {
```
Comment (kwonye) [inline WordPress/Classes/Models/Notifications/Notification+CoreDataProperties.swift]: Is there a reason it is split out like this? I find this actually more confusing than it being in `Notification` itself.

**Comment by kwonye** [comment]
What worries me is that it works both locally and on Travis on `develop`, but neither on this branch.

That said, the broken test locally isn't related to Notifications and Travis does seem to be passing all the tests...

Please just address `Notification+CoreDataProperties` and I'll allow the Travis failure.

**Comment by jleandroperez** [comment]
Hey there @kwonye!!

The reason why i've added `Notification+CoreDataProperties` is that it's the way Xcode autogenerates Data Model classes (Core Data Schema).

It's done this way because whenever you update the Data Model itself, the `+CoreDataProperties` extension can be (automatically regenerated) and overwritten, without a single side effect.

Nevertheless, the `Notification` entity will definitely remain the same for a while longer. I've merged both, the extension and the main class.

Ready for another look, thanks again!!!

**Comment by kwonye** [comment]
> The reason why i've added Notification+CoreDataProperties is that it's the way Xcode autogenerates Data Model classes (Core Data Schema).

ORLY? Hmm, then I'm all for keeping that, sorry!

**Comment by kwonye** [comment]
:shipit: the way it is 👍

**Comment by jleandroperez** [comment]
Pure awesomeness!! thanks @kwonye !!!

**Pull request merged**